lib/std/fmt.zig

```diff
@@ -585,6 +592,18 @@ pub fn formatAsciiChar(
     return output(context, @as(*const [1]u8, &c)[0..]);
 }
 
+pub fn formatUtf8Codepoint(
+    c: u32,
```

Comment:
Should be a `u21`.

Reply:
Why?
std.unicode's funcs doesn't use this restriction.

Reply:
> Why?

Its the actual acceptable range of a utf8 encoder.

>  std.unicode's funcs doesn't use this restriction.

Ah, it probably should.

Reply:
Also, a codepoint may be in a struct's u32 field, so uglifed cast will be needed.

Reply:
> Ah, it probably should.

Done in https://github.com/ziglang/zig/pull/3987